ArchiSteamFarm/GitHub.cs

```diff
@@ -0,0 +1,132 @@
+	internal static class GitHub {
+		internal static async Task<List<ReleaseResponse>> GetNLatestReleases(uint n) {
+			if(n == 0) {
+				ASF.ArchiLogger.LogNullError(nameof(n));
+				return null;
+			}
+
+			string releaseURL = SharedInfo.GithubReleaseURL + "?per_page=" + n;
+
+			return await GetReleasesFromURL(releaseURL).ConfigureAwait(false);
+		}
+
+		internal static async Task<ReleaseResponse> GetLatestRelease(bool stable = true) {
+			string releaseURL = SharedInfo.GithubReleaseURL + (stable ? "/latest" : "?per_page=1");
+
+			if (stable) {
+				return await GetReleaseFromURL(releaseURL).ConfigureAwait(false);
+			}
+			else {
+				return (await GetReleasesFromURL(releaseURL).ConfigureAwait(false))?.ElementAt(0);
+			}
+		}
+
+		internal static async Task<ReleaseResponse> GetRelease(string version) {
+			if(string.IsNullOrEmpty(version)) {
+				ASF.ArchiLogger.LogNullError(nameof(version));
+				return null;
+			}
+
+			return await GetReleaseFromURL(SharedInfo.GithubReleaseURL + "/tags/" + version).ConfigureAwait(false);
+		}
+
+		private static async Task<ReleaseResponse> GetReleaseFromURL(string releaseURL) {
+			if (string.IsNullOrEmpty(nameof(releaseURL))) {
+				ASF.ArchiLogger.LogNullError(nameof(releaseURL));
+				return null;
+			}
+
+			WebBrowser.ObjectResponse<ReleaseResponse> objectResponse = await Program.WebBrowser.UrlGetToJsonObject<ReleaseResponse>(releaseURL).ConfigureAwait(false);
+			if (objectResponse?.Content == null) {
+				ASF.ArchiLogger.LogNullError(nameof(objectResponse) + " || " + nameof(objectResponse.Content));
+				return null;
+			}
+
+			return objectResponse.Content;
+		}
+
+		private static async Task<List<ReleaseResponse>> GetReleasesFromURL(string releaseURL) {
+			if (string.IsNullOrEmpty(nameof(releaseURL))) {
+				ASF.ArchiLogger.LogNullError(nameof(releaseURL));
+				return null;
+			}
+
+			WebBrowser.ObjectResponse<List<ReleaseResponse>> objectResponse = await Program.WebBrowser.UrlGetToJsonObject<List<ReleaseResponse>>(releaseURL).ConfigureAwait(false);
+			if ((objectResponse?.Content == null) || (objectResponse.Content.Count == 0)) {
+				ASF.ArchiLogger.LogNullError(nameof(objectResponse) + " || " + nameof(objectResponse.Content));
+				return null;
+			}
+
+			return objectResponse.Content;
+		}
+
+		[SuppressMessage("ReSharper", "ClassCannotBeInstantiated")]
+		internal sealed class ReleaseResponse {
+			[JsonProperty(PropertyName = "assets", Required = Required.Always)]
+			internal readonly HashSet<Asset> Assets;
+
+			[JsonProperty(PropertyName = "body", Required = Required.Always)]
+			internal readonly string ReleaseNotesInMarkdown;
+
+			[JsonProperty(PropertyName = "tag_name", Required = Required.Always)]
+			internal readonly string Tag;
+
+			[JsonProperty(PropertyName = "body", Required = Required.Always)]
+			internal readonly string MarkdownBody;
+
+			[JsonProperty(PropertyName = "published_at", Required = Required.Always)]
+			internal readonly string PublishDate;
```

Comment:
This should be `DateTime` and `PublishedAt`, not `PublishedDate` (DateTime is self-descriptive).